peripherals/hsem/hsem_wl.yaml

```diff
@@ -0,0 +1,89 @@
+
+"HSEM":
+  "HSEM_R[0123456789]":
+    LOCK:
+      _read:
+        Free: [0, "Semaphore is free"]
+        Locked: [1, "Semaphore is locked"]
+      _write:
+        Free: [0, "Free semaphore"]
+        TryLock: [1, "Try to lock semaphore"]
+    MASTERID: [0, 0xF]
+    PROCID: [0, 0xFF]
+  "HSEM_R1[012345]":
+    LOCK:
+      _read:
+        Free: [0, "Semaphore is free"]
+        Locked: [1, "Semaphore is locked"]
+      _write:
+        Free: [0, "Free semaphore"]
+        TryLock: [1, "Try to lock semaphore"]
+    MASTERID: [0, 0xF]
+    PROCID: [0, 0xFF]
+
+  "HSEM_RLR[0123456789]":
+    LOCK:
+      _read:
+        Free: [0, "Semaphore is free"]
+        Locked: [1, "Semaphore is locked"]
+      _write:
+        Free: [0, "Free semaphore"]
+        TryLock: [1, "Try to lock semaphore"]
+    MASTERID: [0, 0xF]
+    PROCID: [0, 0xFF]
+  "HSEM_RLR1[012345]":
+    LOCK:
+      _read:
+        Free: [0, "Semaphore is free"]
+        Locked: [1, "Semaphore is locked"]
+      _write:
+        Free: [0, "Free semaphore"]
+        TryLock: [1, "Try to lock semaphore"]
+    MASTERID: [0, 0xF]
+    PROCID: [0, 0xFF]
+
+
+  HSEM_IER:
+    "ISE[0123456789]":
+      Disabled: [0, "Interrupt generation disabled"]
+      Enabled: [1, "Interrupt generation enabled"]
+    "ISE1[012345]":
+      Disabled: [0, "Interrupt generation disabled"]
+      Enabled: [1, "Interrupt generation enabled"]
+
+  HSEM_ICR:
+    "ISC[0123456789]":
+      _read:
+        NoEffect: [0, "Always reads 0"]
+      _write:
+        NoEffect: [0, "Interrupt semaphore x status ISFx and masked status MISFx not affected"]
+        Clear : [1, "Interrupt semaphore x status ISFx and masked status MISFx cleared"]
+    "ISC1[012345]":
+      _read:
+        NoEffect: [0, "Always reads 0"]
+      _write:
+        NoEffect: [0, "Interrupt semaphore x status ISFx and masked status MISFx not affected"]
+        Clear : [1, "Interrupt semaphore x status ISFx and masked status MISFx cleared"]
+
+  HSEM_ISR:
+    "ISF[0123456789]":
+      NotPending: [0, "No interrupt pending"]
+      Pending: [1, "Interrupt pending"]
+    "ISF1[012345]":
+      NotPending: [0, "No interrupt pending"]
+      Pending: [1, "Interrupt pending"]
+
+  HSEM_MISR:
+    "MISF[0123456789]":
+      NotPending: [0, "No interrupt pending after masking"]
+      Pending: [1, "Interrupt pending after masking"]
+    "MISF1[012345]":
+      NotPending: [0, "No interrupt pending after masking"]
+      Pending: [1, "Interrupt pending after masking"]
+
+  HSEM_CR:
+    KEY: [0, 0xFFFF]
+    MASTERID: [0, 0xF]
+
+  HSEM_KEYR:
+    KEY: [0, 0xFFFF]
```

Comment:
I think this is a great suggestion. The only thing I would change is to call the core C1 instead of C0. C1 refers to the M4 core in the dual-core version, so if we call the WLE core also C1, only the C2 part of the code needs to be ported or put under feature gates. Makes sense?